src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs

```diff
@@ -338,6 +332,14 @@ private static ParameterDescriptor CreateParameterDescriptor(ParameterInfo param
         var defaultErrorType = errorMetadata?.Type ?? typeof(void);
         var contentTypes = new MediaTypeCollection();
 
+        // If the return type is an IResult or an awaitable IResult, then we should treat it as a void return type
```

Comment:
How are we detecting that "awaitable IResult"?  (Does that mean something other than `Task<IResult>`?)

Reply:
It's detected the code generation layer that inserts the metadata. In this case, there's a slight distinciton in behavior.

When using static code gen (RDG), minimal APIs only detects awaitable types that are either `ValueTask` or `Task`.

When using dynamic code gen (RDF), we used the `CoercedAwaitableInfo` API which uses reflection to check if the target type implements all the required properties for an awaitable type.

https://github.com/dotnet/aspnetcore/blob/27f2a011a4211118552dfb8f38d36e8629267d2b/src/Http/Http.Extensions/src/RequestDelegateFactory.cs#L1027-L1038

Reply:
As we discussed offline, `responseType` has already been unwrapped. 👍